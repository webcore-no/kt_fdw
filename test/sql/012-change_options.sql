set client_min_messages=ERROR;
drop server if exists kt_server cascade;
set client_min_messages=NOTICE;


create server kt_server foreign data wrapper kt_fdw
options(host 'no_hostname', port '42', timeout '-1', reconnect '1');
create foreign table kt_test (key bytea, value bytea, flag text) server kt_server;
create user mapping for public server kt_server;
delete from kt_test;
select pg_sleep(.1);
select * from kt_test;


select pg_sleep(.1);
alter server kt_server options( set host '127.0.0.1');
delete from kt_test;
select pg_sleep(.1);
select * from kt_test;


select pg_sleep(.1);
alter server kt_server options( set port '1978');
delete from kt_test;
select pg_sleep(.1);
select * from kt_test;

select pg_sleep(.1);
alter server kt_server options( set host 'localhouse');
delete from kt_test;
select pg_sleep(.1);
select * from kt_test;

