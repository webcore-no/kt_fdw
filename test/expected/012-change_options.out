set client_min_messages=ERROR;
drop server if exists kt_server cascade;
set client_min_messages=NOTICE;
create server kt_server foreign data wrapper kt_fdw
options(host 'no_hostname', port '42', timeout '-1', reconnect '1');
create foreign table kt_test (key bytea, value bytea, flag text) server kt_server;
create user mapping for public server kt_server;
delete from kt_test;
ERROR:  Failed to connect to no_hostname:42
select pg_sleep(.1);
 pg_sleep 
----------
 
(1 row)

select * from kt_test;
ERROR:  Failed to connect to no_hostname:42
select pg_sleep(.1);
 pg_sleep 
----------
 
(1 row)

alter server kt_server options( set host '127.0.0.1');
delete from kt_test;
ERROR:  Failed to connect to 127.0.0.1:42
select pg_sleep(.1);
 pg_sleep 
----------
 
(1 row)

select * from kt_test;
ERROR:  Failed to connect to 127.0.0.1:42
select pg_sleep(.1);
 pg_sleep 
----------
 
(1 row)

alter server kt_server options( set port '1978');
delete from kt_test;
select pg_sleep(.1);
 pg_sleep 
----------
 
(1 row)

select * from kt_test;
WARNING:  [network error(6)]:unexpected error
 key | value | flag 
-----+-------+------
(0 rows)

select pg_sleep(.1);
 pg_sleep 
----------
 
(1 row)

alter server kt_server options( set host 'localhouse');
delete from kt_test;
WARNING:  [network error(6)]:unexpected error
ERROR:  Failed to connect to localhouse:1978
select pg_sleep(.1);
 pg_sleep 
----------
 
(1 row)

select * from kt_test;
ERROR:  Failed to connect to localhouse:1978
